reflect.go

```diff
@@ -158,10 +158,9 @@ type FieldOptions struct {
 // Usage:
 // `wire.RegisterInterface((*MyInterface1)(nil), nil)`
 func (cdc *Codec) RegisterInterface(ptr interface{}, opts *InterfaceOptions) {
-	cdc.mtx.Lock()
-	defer cdc.mtx.Unlock()
 
```

Comment:
We need these locks.

Reply:
the locks are called on the sub routines so this was deadlocking ...

Reply:
You're right!